### PR TITLE
resolve with new split, not old one

### LINKED ISSUE #982: Adjustment "Add number dimension" doesn't work
Reproduce:
1. Go to https://turnilo.app/#covid19/
2. Change Visualization with selector to Line Chart, you will see possible adjustemnts
3. Pick any adjustment below first i.e. Add a split on <numeric dimension>
Expected result:
Added split
What happens:
Nothing, no split added.

What's interesting, it is impossible to add numeric dimension with "+" menu.

If on Totals Visualization, adding split on <numeric dimension> works fine

### PR DESCRIPTION
**Pull request by adrianmroz-allegro**
Fixes #982

### CODE REVIEW

**Review by kzadurska — APPROVED** [review]
From issue description itself I thought this would be a bigger change 😅 I wonder whether we'd like to have some e2e tests for such cases.

**Comment by adrianmroz-allegro** [comment]
>  I wonder whether we'd like to have some e2e tests for such cases.

We should. But I'm not sure yet how to add comprehensive e2e tests for this. Under the hood, resolvers have quite complex control flow and I'd like to have something better than n*m cases that barely differ.

Maybe I'll start with adding this one edge case and refactor it later ...

File: src/common/visualization-manifests/line-chart/line-chart.ts [hunk 1]
```diff
@@ -90,13 +90,27 @@ const rulesEvaluator = visualizationDependentEvaluatorBuilder
...
     const colorDimension = findDimensionByName(dataCube.dimensions, colorSplit.reference);
-
     const newColorSplit = adjustColorSplit(colorSplit, colorDimension, series, appSettings.customization.visualizationColors);
 
     return Resolve.automatic(8, {
       splits: new Splits({ splits: List([newColorSplit, newTimeSplit]) })
     });
   })
+  .when(Predicates.areExactSplitKinds("number", "*"))
+  .then(({ splits, series, dataCube, appSettings }) => {
```
Comment (adrianmroz-allegro) [inline src/common/visualization-manifests/line-chart/line-chart.ts:100]: Same logic as in `then` from line 85. With small adjustments between time and number split types.

Maybe we could extract some common code but i think it's futile. Let's push forward refactor of visualisation controls and remove resolvers altogether.

**Review by kzadurska — APPROVED** [review]

**Review by mkuthan — APPROVED** [review]
